### PR TITLE
docs(api-javascript): update transformIndexHtml types

### PR DESCRIPTION
**Pull request by Franqsanz**
the documentation specified in the problem has been updated: #602

### CODE REVIEW

File: docs/config/shared-options.md [hunk 1]
```diff
@@ -264,6 +264,8 @@ Si se configura esta opción, los preprocesadores CSS se ejecutarán en los work
 ## css.devSourcemap
 
 - **Experimental** [Hacer Comentarios](https://github.com/vitejs/vite/discussions/13845)
+=======
+- **Experimental** [Hacer comentarios](https://github.com/vitejs/vite/discussions/13845)
```
Comment (mrcego) [inline docs/config/shared-options.md:268]: What happened here?

File: docs/config/shared-options.md [hunk 2]
```diff
@@ -274,6 +276,9 @@ Habilita los mapas de origen durante el desarrollo.
 - **Experimental** [Hacer Comentarios](https://github.com/vitejs/vite/discussions/13835)
 - **Type:** `'postcss' | 'lightningcss'`
 - **Default:** `'postcss'`
+=======
```
Comment (mrcego) [inline docs/config/shared-options.md:279]: And here?

File: docs/guide/build.md [hunk 3]
```diff
@@ -241,6 +241,8 @@ El modo librería incluye una configuración simple y pragmática para librería
 ## Opciones avanzadas para Base
 
 ::: warning
+=======
+:::warning Nota
```
Comment (mrcego) [inline docs/guide/build.md:245]: And here?

**Review by mrcego — CHANGES_REQUESTED** [review]
I don't know what happened, but the merge from main to patch-1 is dangerous in some manners. Please, do a full review about what you're commiting.

Thanks.

**Comment by Franqsanz** [comment]
Hello, the truth is that I don't know what happened, it's some problem with the branch, I'm investigating.

**Comment by mrcego** [comment]
> Hello, the truth is that I don't know what happened, it's some problem with the branch, I'm investigating.

Let me know when you're sure these changes are ready. 

Please, tag every fixed issue from commit.

Thanks in advance.

**Review by Franqsanz — COMMENTED** [review]
I don't see anything wrong with these files, I have already deleted the ones "====" but the rest is fine as far as I can see.

**Comment by Franqsanz** [comment]
It seems to be ok now, if there are still problems close the PR and I will open a new one.

File: docs/config/shared-options.md [hunk 4]
```diff
@@ -263,6 +263,7 @@ Si se configura esta opción, los preprocesadores CSS se ejecutarán en los work
 
 ## css.devSourcemap
 
+- **Experimental** [Hacer Comentarios](https://github.com/vitejs/vite/discussions/13845)
```
Comment (mrcego) [inline docs/config/shared-options.md:266]: The same line keeps repeated here.

File: docs/config/shared-options.md [hunk 5]
```diff
@@ -272,6 +273,8 @@ Habilita los mapas de origen durante el desarrollo.
 ## css.transformer
 
 - **Experimental** [Hacer Comentarios](https://github.com/vitejs/vite/discussions/13835)
+- **Type:** `'postcss' | 'lightningcss'`
```
Comment (mrcego) [inline docs/config/shared-options.md:276]: And here

File: docs/config/shared-options.md [hunk 6]
```diff
@@ -272,6 +273,8 @@ Habilita los mapas de origen durante el desarrollo.
...
+- **Default:** `'postcss'`
```
Comment (mrcego) [inline docs/config/shared-options.md:277]: Here too

**Review by mrcego — CHANGES_REQUESTED** [review]

**Comment by Franqsanz** [comment]
I have already removed the repeated items, this was my mistake, apologies for the loss of time. Thanks for your patience!

**Comment by mrcego** [comment]
> I have already removed the repeated items, this was my mistake, apologies for the loss of time. Thanks for your patience!

Thanks!!

**Pull request merged**